das/coordinator.go

```diff
@@ -76,6 +81,12 @@ func (sc *samplingCoordinator) run(ctx context.Context, cp checkpoint) {
 			}
 		case res := <-sc.resultCh:
 			sc.state.handleResult(res)
+
+			// totalSampledFromWorker is the amount of successfully
+			// sampled headers from the worker
+			totalSampledFromWorker := int64(res.To-res.From) - int64(len(res.failed))
+			sc.metrics.recordTotalSampled(ctx, totalSampledFromWorker)
```

Comment:
Async metrics could only be observed inside metrics Callback. Consider storing a int64 value inside metrics struct and observe it on callback.

Reply:
👍🏻